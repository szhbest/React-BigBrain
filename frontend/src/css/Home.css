.mainPage {
    width: 80%;
    height: 600px;
    text-align: center;
    background: url(../bgc/bgc10.gif);
    background-size: 100% 100%;
    color: #fff;
    margin: 0 auto;
}

.mainPage h1, .BTN {
    font-size: 40px;
    line-height: 70px;
}

.mainPage h1 {
    padding-top: 100px;
}

.BTN {
    color: #fff;
    /* height: 40px; */
    line-height: 70px;
    font: bolder 25px 'Josefin Sans';
    border-radius: 5px;
    background: transparent;
    padding: 15px;
}

.BTN:hover {
    cursor: pointer;
    color: #312e37;
    background-color: white;
}

@media screen and (max-width: 870px) {
    .mainPage {
        width: 100%;
    }
}