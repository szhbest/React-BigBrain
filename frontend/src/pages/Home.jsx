import React from 'react';
import { useHistory } from 'react-router-dom';
import '../css/Home.css';

function Home() {
  const history = useHistory();
  const handleClickL = () => {
    history.push('/login');
  };
  const handleClickR = () => {
    history.push('/register');
  };

  return (
    <div className="mainPage">
      <h1>
        Welcome!
      </h1>
      <button className="BTN B1" onClick={handleClickL} type="button">
        Login
      </button>
      <br />
      <button className="BTN B2" onClick={handleClickR} type="button">
        Register
      </button>
    </div>
  );
}

export default Home;
