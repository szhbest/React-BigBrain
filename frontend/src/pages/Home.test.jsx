import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import Home from './Home';

configure({ adapter: new Adapter() });

describe('Test the login component', () => {
  const wrapper = shallow(<Home />);
  it('should match the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('check H1', () => {
    expect(wrapper.find('h1').text()).toEqual('Welcome!');
  });

  it('contains a button with the "Login"', () => {
    expect(wrapper.find('.B1').text()).toEqual('Login');
  });

  it('contains a button with the "Register"', () => {
    expect(wrapper.find('.B2').text()).toEqual('Register');
  });

  // it('should call the login button on button click', () => {
  //   const onClick = jest.fn(() => console.log(1));
  //   wrapper.find('.B1').simulate('click', onClick);
  //   expect(onClick).toHaveBeenCalledTimes(1);
  // });

  // it('should call the register button on button click', () => {
  //   const pushSpy = jest.spyOn(history, 'push');
  //   wrapper.find('.B2').simulate('click');
  //   expect(pushSpy).toHaveBeenCalled();
  // });
});
