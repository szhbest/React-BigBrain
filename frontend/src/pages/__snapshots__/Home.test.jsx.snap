// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the login component should match the snapshot 1`] = `"<div class=\\"mainPage\\"><h1>Welcome!</h1><button class=\\"BTN B1\\" type=\\"button\\">Login</button><br/><button class=\\"BTN B2\\" type=\\"button\\">Register</button></div>"`;
